pkg/uki/common.go

```diff
@@ -153,3 +159,79 @@ func copyFile(src, dst string) error {
 
 	return destinationFile.Close()
 }
+
+// checkArtifactsignatureIsValid checks that a give efi artifact is signed properly with a signature that would allow it to
+// boot correctly in the current node if secureboot is enabled
+func checkArtifactsignatureIsValid(fs v1.FS, dir string, artifact string, logger sdkTypes.KairosLogger) error {
+	var err error
+	fullArtifact := filepath.Join(dir, artifact)
+	logger.Logger.Info().Str("what", fullArtifact).Msg("Checking artifact")
+	o, err := fs.Open(fullArtifact)
+	if errors.Is(err, os.ErrNotExist) {
+		logger.Warn("%s does not exist", fullArtifact)
+		return fmt.Errorf("%s does not exist", fullArtifact)
+	} else if errors.Is(err, os.ErrPermission) {
+		logger.Warn("%s permission denied. Can't read file", fullArtifact)
+		return fmt.Errorf("%s permission denied. Can't read file", fullArtifact)
+	}
+
+	ok, err := checkValidEfiHeader(o)
+	_ = o.Close()
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to read file %s: %s", fullArtifact, err))
+		return fmt.Errorf("failed to read file %s: %s", fullArtifact, err)
+	}
+	if !ok {
+		logger.Error("invalid pe header")
+		return fmt.Errorf("invalid pe header")
+	}
+
+	// We need to read the current db database to have the proper certs to check against
+	db, err := efi.Getdb()
```

Comment:
Ideally, we would need to check that the certificate signing the efi file is not blacklisted in dbx and that the binary signature is not blacklisted either. Even worse, we would need to check that the KEK and PK certs that sign the db that signs the efi file, are not blacklisted. This last one though, [I'm having trouble verifying](https://github.com/kairos-io/kairos/issues/2429).

Reply:
good point, dbx should be checked to see if the cert is not in there.

Regarding the PK/KEK, can they be blacklisted but not the db? I mean, db+dbx is the way of checking the signatures in files no? Any other verification, needs to be done by the firmware itself to allow you updating db/dbx no?

In any case, doesnt look like sbctl is actually checking the dbx/pk/kek either, only checks if its signed with the db cert....